uPortal-api/uPortal-api-internal/src/main/java/org/apereo/portal/portlets/favorites/FavoritesUtils.java

```diff
@@ -285,4 +284,29 @@ public boolean hasAnyFavorites(IUserLayout layout) {
         return (!getFavoritePortletLayoutNodes(layout).isEmpty()
                 || !getFavoriteCollections(layout).isEmpty());
     }
+
+    /**
+     * A helper method introduced to filter out duplicates by key
+     *
+     * @param keyExtractor
+     * @param <T>
+     * @return
+     */
+    public static <T> Predicate<T> distinctByKey(Function<? super T, Object> keyExtractor) {
+        Map<Object, Boolean> map = new ConcurrentHashMap<>();
+        return t -> map.putIfAbsent(keyExtractor.apply(t), Boolean.TRUE) == null;
+    }
+
+    public static List<IUserLayoutNodeDescription> filterFavoritesToUnique(
+            List<IUserLayoutNodeDescription> originalFavoritesList) {
+        List<IUserLayoutNodeDescription> uniqueFavorites = new ArrayList<>();
+        Predicate<IUserLayoutNodeDescription> predicate;
+        predicate = FavoritesUtils.distinctByKey(p -> p.getName());
+        for (IUserLayoutNodeDescription favorite : originalFavoritesList) {
+            if (predicate.test(favorite)) {
+                uniqueFavorites.add(favorite);
+            }
+        }
+        return uniqueFavorites;
+    }
```

Comment:
You may want to look at this example to make `filterFavoritesToUnique()` a little more tidy:
https://howtodoinjava.com/java8/java-stream-distinct-examples/

```
List<IUserLayoutNodeDescription> uniqueFavorites = originalFavoritesList.stream()
  .filter( FavoritesUtils.distinctByKey(p -> p.getName() )
  .collect( Collectors.toList() );
```

Reply:
Thanks for this feedback, I refactored the filterFavoritesToUnique method to follow the example you provided.